src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -1186,4 +1182,94 @@ private void sendBulkUploadNotification(String orgId, String orgName, String fil
 				serverConfig.getSbUrl() + serverConfig.getSbSendNotificationEmailPath(), request,
 				ProjectUtil.getDefaultHeaders());
 	}
+
+	@Override
+	public SBApiResponse getNotificationPreferencesById(String userId) {
+		SBApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_READ_NOTIFICATION_PREFERENCE);
+		String errMsg = null;
+		if (StringUtils.isEmpty(userId)) {
+			response.getParams().setErrmsg("Please Provide a user Id");
+			response.setResponseCode(HttpStatus.BAD_REQUEST);
+			response.getParams().setStatus(Constants.FAILED);
+			return response;
+		}
+		Map<String, Object> request = new HashMap<>();
+		request.put(Constants.USER_ID, userId);
+		try {
+			List<Map<String, Object>> notificationPreferences = cassandraOperation.getRecordsByProperties(
+					Constants.KEYSPACE_SUNBIRD, Constants.TABLE_USER_NOTIFICATION_PREFERENCE, request,
+					Collections.singletonList(Constants.NOTIFICATION_PREFERENCE));
+			GsonBuilder builder = new GsonBuilder();
+			Gson gson = builder.create();
+			List<Map<String, Object>> mapping = gson.fromJson(notificationPreferences.toString(), List.class);
```

Comment:
notificationPreferences -- is already list of Map... why are you converting it to string and then converting back to list ?